file_process/csv.py

```diff
@@ -5,67 +5,65 @@
-            data_stream.seek(0)
+    def __init__(self, file: BytesIO, **kwargs):
+        self.delimiter = kwargs.get('delimiter') or self._get_delimiter(file)
         try:
-            df = pd.read_csv(data_stream, sep=delimiter)
+            self.data_df = pd.read_csv(file, sep=self.delimiter)
         except ParserError as exc:
             raise DelimiterError() from exc
-        return df
 
-    def get_preview_data(self, df: pd.DataFrame):
-        var_names = list(df.columns)
-        obs_preview = df.head(min(10, df.shape[0]))
-        return var_names, obs_preview
+    def _get_delimiter(self, file: BytesIO):
+        reader = pd.read_csv(file, sep=None, iterator=True, nrows=10)
+        delimiter = reader._engine.data.dialect.delimiter  # pylint: disable=protected-access
+        file.seek(0)
+        return delimiter
 
-    def model_file_validation(self, df: pd.DataFrame, model_metadata_file: BytesIO, need_target: bool = True):
+    def get_obs(self):
+        return self.data_df.head(min(10, self.data_df.shape[0]))
+
+    def get_var_names(self):
+        return list(self.data_df.columns)
+
+    def get_preview(self):
+        var_names = self.get_var_names()
+        obs_preview = self.get_obs()
+        return var_names, self.create_tabular_response(obs_preview), None
+
+    def validate(self):
+        pass
+
+    def model_file_validation(self, model_metadata_file: BytesIO):
         reader = json.load(model_metadata_file)
         var_names = set(reader['columns'])
         target_names = set(reader['targets'])
         metadata = reader.get('metadata', {})
-        dataset_vars = set(df.columns)
+        dataset_vars = set(self.data_df.columns)
 
-        if need_target:
-            all_targets = metadata.get('require_all_targets', True)
-            if all_targets == 'all':
-                are_targets_valid = not target_names or all(elem in dataset_vars for elem in target_names)
-            elif all_targets == 'some':
-                are_targets_valid = not target_names or any(elem in dataset_vars for elem in target_names)
-            else:
-                are_targets_valid = True
+        all_targets = metadata.get('require_all_targets', True)
+        if all_targets == 'all':
+            are_targets_valid = not target_names or all(elem in dataset_vars for elem in target_names)
             if not are_targets_valid:
-                raise ModelFileValidationTargetsError
+                raise NotAllTargetsError
+        elif all_targets == 'some':
+            are_targets_valid = not target_names or any(elem in dataset_vars for elem in target_names)
+            if not are_targets_valid:
+                raise NotSomeTargetsError
+
         are_variables_valid = all(elem in dataset_vars.difference(target_names)
                                   for elem in var_names.difference(target_names))
         if not are_variables_valid:
             raise ModelFileValidationVariablesError
```

Comment:
If we can tell user what variable failed, that will be helpful

Reply:
that is a good point